### PR TITLE
Add libraw/0.19.5

### PR DESCRIPTION
**Pull request by p-podsiadly**
**libraw/0.19.5**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libraw/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    default_options = {"shared": False, "fPIC": True}
+
+    requires = [
+        "libjpeg/9d",
+        "lcms/2.10"
+    ]
+
+    version = "0.19.5" # TODO remove
+
+    _source_subfolder = "source_subfolder"
```
Comment (prince-chrismc) [inline recipes/libraw/all/conanfile.py:24]:
```suggestion
    @property
    def _source_subfolder(self):
        return "source_subfolder"
```

File: recipes/libraw/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    settings = "os", "arch", "compiler", "build_type"
+
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    requires = [
+        "libjpeg/9d",
+        "lcms/2.10"
+    ]
+
```
Comment (prince-chrismc) [inline recipes/libraw/all/conanfile.py:21]:
```suggestion
    _cmake = None
```

File: recipes/libraw/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("LibRaw-" + self.version, self._source_subfolder)
+
+    def configure_cmake(self):
+        cm = CMake(self)
+        
+        cm.configure()
+        return cm
```
Comment (prince-chrismc) [inline recipes/libraw/all/conanfile.py:38]:
```suggestion
    def _configure_cmake(self):
        if self._cmake:
            return self._cmake
        self._cmake = CMake(self)
        self._cmake.configure()
        return self._cmake
```

caching the cmake object avoid a second configure making the builds faster

File: recipes/libraw/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    def build(self):
+        self.configure_cmake().build()
```
Comment (prince-chrismc) [inline recipes/libraw/all/conanfile.py:41]:
```suggestion
        cmake = self._configure_cmake()
        cmake.build()
```

File: recipes/libraw/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    def package(self):
+        self.configure_cmake().install()
+        self.copy("LICENSE.*", src=self._source_subfolder, dst="licenses")
```
Comment (prince-chrismc) [inline recipes/libraw/all/conanfile.py:45]:
```suggestion
        self.copy("LICENSE.*", src=self._source_subfolder, dst="licenses")
        cmake = self._configure_cmake()
        cmake.install()
```

File: recipes/libraw/all/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,38 @@
+cmake_minimum_required(VERSION 3.1)
+
+project(LibRaw)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
```
Comment (prince-chrismc) [inline recipes/libraw/all/CMakeLists.txt:6]:
```suggestion
conan_basic_setup()
```

File: recipes/libraw/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    requires = [
+        "libjpeg/9d",
+        "lcms/2.10"
+    ]
+
+    version = "0.19.5" # TODO remove
```
Comment (prince-chrismc) [inline recipes/libraw/all/conanfile.py:22]:
```suggestion
```

File: recipes/libraw/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    requires = [
+        "libjpeg/9d",
+        "lcms/2.10"
+    ]
```
Comment (prince-chrismc) [inline recipes/libraw/all/conanfile.py:20]:
```suggestion
    def requirements(self):
        self.requires("libjpeg/9d")
        self.requires("lcms/2.10")
```

File: recipes/libraw/all/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,38 @@
+cmake_minimum_required(VERSION 3.1)
```
Comment (danimtb) [inline recipes/libraw/all/CMakeLists.txt:1]: I guess the project doesn't provide any build files and we have to add it, Anyway, please add a note about it at the top of this file

File: recipes/libraw/all/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    add_definitions(-DWIN32)
+
+    if(BUILD_SHARED_LIBS)
+        add_definitions(-DLIBRAW_BUILDLIB)
+    else()
+        add_definitions(-DLIBRAW_NODLL)
+    endif()
+endif()
+
+add_definitions(-DUSE_JPEG -DUSE_JPEG8)
```
Comment (uilianries) [inline recipes/libraw/all/CMakeLists.txt:29]: I would prefer `target_compile_definitions` because is not global. But I'm fine with this.

**Pull request merged**